src/main/java/com/midaswebserver/midasweb/services/UserServiceImp.java

```diff
@@ -0,0 +1,264 @@
+            return false;
+        }
+        userRepo.save(user);
+        log.debug("add: '{}' has been added to the database", user.getUsername());
+        return true;
+    }
+
+    /**
+     * ValidateUniqueUsername checks to see if there is another person in the database with the same username
+     *
+     * @param username
+     * @return boolean
+     */
+    @Override
+    public boolean validateUniqueUsername(String username) {
+        log.debug("validateUser: '{}' attempted login", getUserByUsername(username));
+        // Always do the lookup in a case-insensitive manner (lower-casing the data).
+        List<User> users = userRepo.findByUsernameIgnoreCase(username);
+
+        // We expect 0 or 1, so if we get more than 1, bail out as this is an error we don't deal with properly.
+        if (users.size() != 0) {
+            log.debug("validateUniqueUsername: found {} users, invalid quantity. Valid = 1", users.size());
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * delete deletes a user from the database
+     *
+     * @param user
+     * @return boolean
+     */
+    @Override
+    public boolean delete(User user) {
+        if (user == null) {
+            log.error("Delete: null was passed to method");
+            return false;
+        }
+        if (this.getUserByUsername(user.getUsername()) == null) {
+            log.error("Delete: User '{}' was not found in database to delete", user.getUsername());
+            return false;
+        }
+        userRepo.delete(user);
+        log.debug("Delete: User '{}' was deleted from the database", user.getUsername());
+        return true;
+    }
+
+
+    /**
+     * Gets User by Id
+     * TODO have validation and safe and expected returns;
+     *
+     * @param ID
+     * @return User if user excists, returns null otherwise and if user passes null ID
+     */
+    @Override
+    public User getUserById(Long ID) {
+        if (ID == null) {
+            return null;
+        }
+        try {
+            return userRepo.findById(ID).get();
+        } catch (Exception e) {
+            log.error("getUserByID: id, '{}', didnt have corresponding user", ID, e);
+        }
+        return null;
+    }
+
+    /**
+     * Gets user with the corresponding name
+     *
+     * @param userName
+     * @return user with corresponding name, or null if no user was found
+     */
+    @Override
+    public User getUserByUsername(String userName) {
+        if (userName != null) {
+            try {
+                List<User> users = userRepo.findByUsernameIgnoreCase(userName);
+                if (users.size() == 1) {
+                    return users.get(0);
+                } else {
+                    log.warn("GetUserByName: 0 or 1 < users found with name '{}'", userName);
+                }
+            } catch (Exception e) {
+                log.info("getUserByName:", e);
```

Comment:
I'd call this an error not an info, and the log message doesn't match the method name, or provide any context on what's going on (such as providing the userName).